src/useObserver.ts

```diff
@@ -25,39 +35,84 @@ export function useObserver<T>(
+        // First render for this component (or first time since a previous
+        // reaction from an abandoned render was disposed).
+
+        const newReaction = new Reaction(observerComponentNameFor(baseComponentName), () => {
+            // Observable has changed, meaning we want to re-render
+            // BUT if we're a component that hasn't yet got to the useEffect()
+            // stage, we might be a component that _started_ to render, but
+            // got dropped, and we don't want to make state changes then.
+            // (It triggers warnings in StrictMode, for a start.)
+            if (trackingData.mounted) {
+                // We have reached useEffect(), so we're mounted, and can trigger an update
+                forceUpdate()
+            } else {
+                // We haven't yet reached useEffect(), so we'll need to trigger a re-render
+                // when (and if) useEffect() arrives.  The easiest way to do that is just to
+                // drop our current reaction and allow useEffect() to recreate it.
+                newReaction.dispose()
+                reactionTrackingRef.current = null
+            }
         })
-    }
 
-    const dispose = () => {
-        if (reaction.current && !reaction.current.isDisposed) {
-            reaction.current.dispose()
-        }
+        const trackingData = createTrackingData(newReaction)
+        reactionTrackingRef.current = trackingData
+        scheduleCleanupOfReactionIfLeaked(reactionTrackingRef)
     }
 
-    useDebugValue(reaction, printDebugValue)
+    const { reaction } = reactionTrackingRef.current!
+    React.useDebugValue(reaction, printDebugValue)
 
-    useUnmount(() => {
-        dispose()
-    })
+    React.useEffect(() => {
+        // Called on first mount only
+        recordReactionAsCommitted(reactionTrackingRef)
+
+        if (reactionTrackingRef.current) {
+            // Great. We've already got our reaction from our render;
+            // all we need to do is to record that it's now mounted,
+            // to allow future observable changes to trigger re-renders
+            reactionTrackingRef.current.mounted = true
+        } else {
+            // The reaction we set up in our render has been disposed.
+            // This is either due to bad timings of renderings, e.g. our
+            // component was paused for a _very_ long time, and our
+            // reaction got cleaned up, or we got a observable change
+            // between render and useEffect
+
+            // Re-create the reaction
+            reactionTrackingRef.current = {
+                reaction: new Reaction(observerComponentNameFor(baseComponentName), () => {
+                    // We've definitely already been mounted at this point
+                    forceUpdate()
+                }),
+                cleanAt: Infinity
+            }
+            forceUpdate()
+        }
+
+        return () => reactionTrackingRef.current!.reaction.dispose()
```

Comment:
This seems to be the line missing for #226
`reactionTrackingRef.current = null;`

Reply:
Hm, was I looking at the wrong line before? 🤔 I suppose I will cut some alpha release with it and try it out. Thanks for spotting it.